src/ripple/protocol/impl/RippleAddress.cpp

```diff
@@ -813,4 +845,62 @@ RippleAddress RippleAddress::createSeedGeneric (std::string const& strText)
     return naNew;
 }
 
+uint256 keyFromSeed (uint128 const& seed)
+{
+    Serializer s;
+
+    s.add128 (seed);
+    uint256 result = s.getSHA512Half();
+
+    s.secureErase ();
+
+    return result;
+}
+
+RippleAddress getSeedFromRPC (Json::Value const& params)
+{
+    bool const has_passphrase = params.isMember (jss::passphrase);
+    bool const has_seed       = params.isMember (jss::seed);
+    bool const has_seed_hex   = params.isMember (jss::seed_hex);
+
+    int const n_secrets = has_passphrase + has_seed + has_seed_hex;
+
+    if (n_secrets > 1)
+    {
+        // `passphrase`, `seed`, and `seed_hex` are mutually exclusive.
+        return RippleAddress();
```

Comment:
Similar to the comment above, instead of returning a default `RippleAddress`, how about returning a `std::pair<bool, RippleAddress>`?